src/persist-txn/src/txn_cache.rs

```diff
@@ -426,6 +444,9 @@ impl<T: Timestamp + Lattice + TotalOrder + StepForward + Codec64> TxnsCacheState
                 data_id.to_string(),
                 ts
             );
+            if ts == compacted_ts {
+                self.unapplied_forgets.push_back((data_id, ts.clone()));
```

Comment:
This PR would be massively simpler if I replaced this line with `self.unapplied_registers.push_back((data_id, ts.clone()));` considering that applying a register is identical to applying a forget.

Any thoughts?

Reply:
Heh, I lean toward combining the two. That was my first thought on seeing the PR, even before I got to this question.

Edit: Oh do we need this to make sure we process registers before batches before forgets in apply? That's actually a great argument for making forget exclusive instead of inclusive (in a sense, register/forget would always happen at the "beginning" of the timestamp). I think even in light of this, I still support my thought above of combining them (and adjusting the comment in apply about doing registers before batches somehow to compensate)

Reply:
> Oh do we need this to make sure we process registers before batches before forgets in apply?

That's why I kept them separate, though I would replace "need" with "probably want to in the future". I think I also agree towards combining register and forget.

Reply:
I just pushed an update to combine them. I kept around the updated unapplied iterator to make it easier to separate them in the future, let me know if you'd prefer I revert that.